packages/filecoin-address/src/index.ts

```diff
@@ -180,31 +199,8 @@ export function newDelegatedEthAddress(
 }
 
 export function decode(address: string): Address {
-  checkAddressString(address)
-
-  const coinType = address.slice(0, 1) as CoinType
-  /* tslint:disable-next-line:radix */
-  const protocol = parseInt(address.slice(1, 2)) as Protocol
-  const raw = address.substring(2, address.length)
-  const protocolByte = leb.unsigned.encode(protocol)
-
-  if (protocol === Protocol.ID) {
-    return newIDAddress(raw, coinType)
-  }
-
-  const payloadChecksum = base32.decode(raw)
-  const length = payloadChecksum.length
-  const payload = payloadChecksum.slice(0, length - 4)
-  const checksum = payloadChecksum.slice(length - 4, length)
-  if (validateChecksum(uint8arrays.concat([protocolByte, payload]), checksum)) {
-    throw Error("Checksums don't match")
-  }
```

Comment:
@Schwartz10 just to be transparent, it was me who broke the `f1`, `f2` and `f3` address validation here. Because you were throwing an error when `validateChecksum` was truthy, which is counter-intuitive, but did work :)